src/e3/anod/deps.py

```diff
@@ -79,8 +79,24 @@ def __init__(
         self.host = host
         self.target = target
         self.build = build
-        self.qualifier = qualifier
         self.local_name = local_name if local_name is not None else name
+
+        self.qualifier: str | None
+        if isinstance(qualifier, dict):
+            for q in set(qualifier.keys()):
```

Comment:
no need to cast on set

Reply:
The cast to set is way to copy it. Otherwise python complains because the size of the dictionary change during the iteration.

Reply:
To me it's ok. Using a `set` or a `list` seems to do the same.

```python
>>> a = {1: "one", 2: "two", 3: "three"}
>>> for k in list(a.keys()):
...     a[1 + len(a)] = "something"
...     print(k)
... 
1
2
3
>>> a
{1: 'one', 2: 'two', 3: 'three', 4: 'something', 5: 'something', 6: 'something'}
>>> for k in a:
...     a[1 + len(a)] = "something"
...     print(k)
... 
1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
>>>
````